pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -325,6 +326,22 @@ func (w *Webhook) inject(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
 	return w.injectAutoInstruConfig(pod, libsToInject, autoDetected, injectionType)
 }
 
+func injectSecurityClientLibraryConfig(pod *corev1.Pod) {
+	injectEnvVarIfConfigKeySet(pod, "admission_controller.auto_instrumentation.asm.enabled", "DD_APPSEC_ENABLED")
+	injectEnvVarIfConfigKeySet(pod, "admission_controller.auto_instrumentation.iast.enabled", "DD_IAST_ENABLED")
+	injectEnvVarIfConfigKeySet(pod, "admission_controller.auto_instrumentation.sca.enabled", "DD_SCA_ENABLED")
+}
+
+func injectEnvVarIfConfigKeySet(pod *corev1.Pod, configKey string, envVarKey string) {
```

Comment:
I think we need a comment with what you explained in another comment in this PR.

I would set the _ENABLED envs only if the corresponding option is set to true. It'd be good to explain why that doesn't work in this case.

Reply:
added comment in 675b22f098